xtask/src/cli.rs

```diff
@@ -0,0 +1,60 @@
+use std::process::exit;
+
+use anyhow::{anyhow, bail, Context};
+use pico_args::Arguments;
+
+const HELP: &str = "\
+Usage: xtask <COMMAND>
+
+Commands:
+  run-wasm
+
+Options:
+  -h, --help  Print help
+";
+
+pub(crate) struct Args {
+    pub(crate) subcommand: Subcommand,
+}
+
+impl Args {
+    pub fn parse() -> Self {
+        let mut args = Arguments::from_env();
+        log::debug!("parsing args: {args:?}");
+        if args.contains("--help") {
+            eprint!("{HELP}");
+            exit(101);
+        }
+        match (|| -> anyhow::Result<_> {
+            let subcommand = Subcommand::parse(args)?;
+            Ok(Self { subcommand })
+        })() {
+            Ok(this) => this,
+            Err(e) => {
+                eprintln!("{:?}", anyhow!(e));
+                exit(1)
+            }
+        }
+    }
+}
+
+pub(crate) enum Subcommand {
+    RunWasm {
+        // TODO: I'd really like for this to not leak. `Vec<OsString>` plz?
+        args: Arguments,
+    },
+}
+
+impl Subcommand {
+    fn parse(mut args: Arguments) -> anyhow::Result<Subcommand> {
+        args.subcommand()
+            .context("failed to parse subcommand")
+            .and_then(|parsed| match parsed.as_deref() {
```

Comment:
Could we use regular control flow + ? here (or let-else), it'd be a lot clearer.

Reply:
I would additionally move the argument to the `match` into it's own variable, but beyond that it looks fine.

Reply:
Moved with 9e38358a.